p2p/host.go

```diff
@@ -188,6 +188,12 @@ func (h *host) send(p peer.ID, msg Message) (err error) {
 		return err
 	}
 
+	// variable length encoding
+	_, err = s.Write(common.Uint16ToBytes(uint16(len(encMsg)))[0:1])
```

Comment:
is this LEB128?

Reply:
I am not actually sure if this is officially LEB128 encoding. This is what was here before. It was commented out at some point, then removed per clean up request.

What I do know is that this is what currently works with polkadot and is the solution to the read buffer error I mentioned at some point, `b"": Not enough data to fill buffer`.

Reply:
if you use the LEB encoding function from before that would probably be preferred, for longer length messages the encoding might be >2 bytes, so this wouldn't work